### ISSUE TITLE
Missing file "converter_common"

### ISSUE THREAD

**Issue by vToMy** [mnakamura1337/engine_bgi#1]
Hey,
Trying to use your script and there seems to be a missing "converter_common" file referenced in bgi2story.

**Comment by mnakamura1337** [comment]
Yeah, you're right. It's kind of a common module for all converters, and I still haven't decided where to commit it to :(

Here it is: https://gist.github.com/mnakamura1337/1cd2810ca7837ca27bc6b3420e8cfb6b

I'll try to decide on its final place soon.

**Comment by vToMy** [comment]
Thanks!
You can create a new repository for common tools and reference it via git submodule or subtree.

**Comment by mnakamura1337** [comment]
I'm kind of reluctant, because there are a few directly contradicting ideas here.

First, I'd like to create a system of repositories that will aid porting in a way that would be straight-forward and as flexible as possible (and that means it shouldn't be tied to one particular OS, programming language, platform, my beloved Kaitai Struct, etc, etc). It should be easy for anyone to just start a new port / engine exploration project and start getting results as soon as possible with only the things one knows / wants to use.

But second, for such a system to be really useful to everyone, it would help immersively if all the ports / engines would be written in some kind of single standard. That probably means obligatory use of Kaitai Struct to describe file structures, obligatory documentation, and mandating some other things like that.

First proposal means "start fast, have to solve communication problems later". Second proposal means "start slow (must learn all these standards and tools first), but have a steady & fast continuation of work".

So I'm kind of still hesitating which path to choose.
